blob/service.go

```diff
@@ -0,0 +1,278 @@
+package blob
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/cosmos/cosmos-sdk/types"
+	logging "github.com/ipfs/go-log/v2"
+
+	"github.com/celestiaorg/celestia-app/pkg/shares"
+	apptypes "github.com/celestiaorg/celestia-app/x/blob/types"
+	libhead "github.com/celestiaorg/go-header"
+	"github.com/celestiaorg/nmt/namespace"
+
+	"github.com/celestiaorg/celestia-node/header"
+	"github.com/celestiaorg/celestia-node/share"
+	"github.com/celestiaorg/celestia-node/state"
+)
+
+var (
+	log = logging.Logger("blob")
+
+	errBlobNotFound = errors.New("blob: blob not found")
+)
+
+type Service struct {
+	// accessor dials the given celestia-core endpoint to submit blobs.
+	accessor *state.CoreAccessor
+	// sGetter retrieves the EDS to fetch all shares from the requested header.
+	sGetter share.Getter
+	// hGetter allows to get the requested header.
+	hGetter libhead.Getter[*header.ExtendedHeader]
+	// headGetter gets current network head.
+	headGetter libhead.Head[*header.ExtendedHeader]
+}
+
+func NewService(
+	state *state.CoreAccessor,
+	getter share.Getter,
+	hGetter libhead.Getter[*header.ExtendedHeader],
+	headGetter libhead.Head[*header.ExtendedHeader],
+) *Service {
+	return &Service{
+		accessor:   state,
+		sGetter:    getter,
+		hGetter:    hGetter,
+		headGetter: headGetter,
+	}
+}
+
+func (s *Service) Submit(ctx context.Context, f int64, gasLimit uint64, blobs ...*Blob) (uint64, error) {
```

Comment:
I feel conflicted about making gasLimit and fee-required params. These are not documented in the main issue nor the discussion. Additionally, we discussed in person that we want to do something other than that.

Our interface will and should work properly with the default fee params. It doesn't mean that we should not allow adjusting them, and this is where our beloved options come into place. 

We should remove fee params to comply with the interface we want and implement options in the subsequent PR, leaving this PR free from fees and options discussions. 

After we merge the PR, we should discuss altogether which pattern we want to use for options here(functional or just a structure) as there are tradeoffs between them

Reply:
Took from the discussion: 	
```
// Submit sends PFB transaction and reports the height it landed on.
	// Allows sending multiple Blobs atomically synchronously.
	// Uses default wallet registered on the Node.
	Submit(context.Context, uint64 fee, uint64 gasLimit, Blob...) (height uint64, error)
```

Reply:
Ok, I missed your edit there

Reply:
Figured out, who is that bad guy, who was editing the discussion. will remove these params and use defaults

Reply:
But let's discuss what default values will we use.
Found 
`DefaultGasLimit      = 210000`